### PR TITLE
92 - Execute e2e tests in GitHub action

### LINKED ISSUE #92: Configure e2e test environment for GitHub actions
## Overview of the Feature Request

Currently the e2e tests are temporarily disabled on GitHub actions (the corresponding GitHub action job is commented out), since they need a running dedicated Dataverse instance to work.

The goal is to set up a suitable environment for e2e tests to work on the GitHub action and enable the job to run again.

## What inspired the request?
- Disabling e2e action job on #67 

## What kind of user is the feature intended for?
Developers and testers

## What existing behavior do you want changed?
N/A

## Any brand new behavior do you want to add to Dataverse?
N/A

## Any open or closed issues related to this feature request?
- #67

### PR DESCRIPTION
**Pull request by GPortas**
## What this PR does / why we need it:

Re-enable the execution of E2E tests through GitHub actions.

## Which issue(s) this PR closes:

- Closes #92 

## Special notes for your reviewer:

The environment needs to know which branch of Dataverse to run in the containers.

For now, I've added the branch name as an environment variable within the action's yml file.

The variable has to be updated in the development of features to ensure that e2e executes the right Dataverse branch. 

We may need to improve this mechanism in the future to make it more practical.

## Suggestions on how to test this:

See how the action is executed in this PR.

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:

No

## Is there a release notes update needed for this change?:

No

## Additional documentation:

None

### CODE REVIEW

**Review by pdurbin — COMMENTED** [review]
I'm excited it's working but let's talk about the `9588-datasets-api-extension` branch being hard coded.

File: .github/workflows/test.yml
Comment (GPortas) [reply .github/workflows/test.yml:6]: The problem is that the integration tests must point to the Dataverse backend branch (image tag) used by the specific version of the js-dataverse package used, otherwise they could fail, not having the necessary Dataverse API changes.

We could set develop as you suggest (seems like a cleaner solution), but then we need to make sure to merge backend PRs first. This way the frontend PRs should wait (before accepting them) for the backend PRs to be merged for the e2e action to pass.

Comment (donsizemore) [reply .github/workflows/test.yml:6]: is there a way to pass repo and branch variables with the commit message?

Comment (GPortas) [reply .github/workflows/test.yml:6]: We could read the branch name of the commit message from the action via `${{ github.event.head_commit.message }}` and detect if a branch name has been sent (e.g. surrounded the branch name by some text markers or using regex). In that case, we use that tag in the action. Otherwise, we use develop. What do you think?

Comment (donsizemore) [reply .github/workflows/test.yml:6]: It's what dataverse-ansible does (check for specified repo / branch, else assume defaults)

Comment (pdurbin) [reply .github/workflows/test.yml:6]: Back to the order of PRs being merged... recently we merged [a frontend PR](https://github.com/IQSS/dataverse-frontend/pull/112) and [a JS PR](https://github.com/IQSS/dataverse-client-javascript/pull/60) before we merged the [backend PR](https://github.com/IQSS/dataverse/pull/9592) they depend on. This is backwards. The backend PR should be merged first. Otherwise, the frontend and js work will have to be revisited if changes are required in the backend PR. I also mentioned this here: https://iqss.slack.com/archives/C010LA04BCG/p1686690116229119?thread_ts=1686603802.872139&cid=C010LA04BCG

**Comment by pdurbin** [comment]
I'm seeing...

"npm ERR! code E400
npm ERR! 400 Bad Request - GET https://npm.pkg.github.com/IQSS/axe-playwright

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2023-07-17T13_54_07_297Z-debug-0.log
Error: Process completed with exit code 1."

... https://github.com/IQSS/dataverse-frontend/actions/runs/5576533857/jobs/10187960772

I'm going to try clicking "re-run all jobs".

**Comment by pdurbin** [comment]
After rerunning e2e it passed. It seems there might be some flakiness due to npm but let's keep an eye on it.

**Review by pdurbin — APPROVED** [review]
I'm excited to see these e2e tests! Approved!

**Comment by kcondon** [comment]
Tests pass, will merge once conflicts are resolved, @GPortas

**Comment by pdurbin** [comment]
@GPortas can you please resolve the conflicts in this PR?

**Comment by GPortas** [comment]
Done. After the merge with develop, I've had to add some fixes unrelated to the PR changes @kcondon

**Pull request merged**
